### PR TITLE
Fix issue when building dependency graph for files with circular refs

### PR DESCRIPTION
**Pull request by zhengbli**

### CODE REVIEW

File: src/compiler/utilities.ts [hunk 1]
```diff
@@ -2272,7 +2272,7 @@ namespace ts {
                 sourceMapFilePath: getSourceMapFilePath(jsFilePath, options),
                 declarationFilePath
             };
-            action(emitFileNames, [sourceFile], /*isBundledEmit*/false);
+            action(emitFileNames, [sourceFile], /*isBundledEmit*/false, emitOnlyDtsFiles);
```
Comment (vladima) [inline src/compiler/utilities.ts]: This (and similar changes around) don't not seem to be related to the original issue

Comment (zhengbli) [inline src/compiler/utilities.ts]: It is actually, because otherwise the emitter would not write the triple slashes in the declaration file

Comment (vladima) [inline src/compiler/utilities.ts]: nit: can you make parameter non-optional?

**Comment by vladima** [comment]
:+1: with a suggestion

**Pull request merged**